### PR TITLE
Seaborn Plots

### LINKED ISSUE #16: Plotly Backend
Use `plotly` for rendering plots.

### LINKED ISSUE #56: Seaborn Plots
Update plotting methods to use seaborn.

### LINKED ISSUE #60: Cast Categorical
If labeling returns discrete values, maybe helpful to automatically cast to the categorical `dtype`. This can help downstream processes such as whether to plot kde, encoding / decoding values, etc.

### PR DESCRIPTION
**Pull request by jeff-hernandez**
This PR uses Seaborn for plotting Label Times. Theses are plot examples for [categorical](https://compose.featurelabs.com/en/seaborn-plots/examples/predict-next-purchase/example.html#Plot-Labels) and [continuous](https://compose.featurelabs.com/en/predict-remaining-useful-life/examples/predict-remaining-useful-life/example.html#Plot-Labels) Label Times. Closes #60. Closes #16 . Closes #56 .

### CODE REVIEW

**Comment by kmax12** [comment]
does count by time make sense for continuous? I feel like it only makes sense if you were to bucket it in discrete labels or if you take some sort of rolling average by time

**Comment by jeff-hernandez** [comment]
I agree. When `count_by_time` is called with continuous labels, not sure whether to:
- make discrete automatically before plotting
- return none
- raise an error

Also, what plot method can be for continuous labels vs. cutoff times?

**Comment by kmax12** [comment]
maybe just to overall total number of label by time. don't try to break it up at all.

**Comment by jeff-hernandez** [comment]
I see. So, we'd just count label per cutoff time without grouping by instance.

**Comment by jeff-hernandez** [comment]
Updated plot for count by time [here](https://compose.featurelabs.com/en/predict-remaining-useful-life/examples/predict-remaining-useful-life/example.html). May need a hard refresh in browser to remove cached files.

**Comment by jeff-hernandez** [comment]
For plotting `count_by_time`, I think it might make sense not to use a cumulative sum and to switch to a histogram. Additionally, if labels are categorical, the histogram can be stacked.

File: composeml/label_times.py [hunk 1]
```diff
@@ -19,55 +35,49 @@ def __init__(self, data=None, name=None, target_entity=None, settings=None, tran
...
     def _constructor(self):
         return LabelTimes
 
     @property
-    def distribution(self):
-        labels = self.assign(count=1)
-        labels = labels.groupby(self.name)
-        distribution = labels['count'].count()
-        return distribution
+    def _is_categorical(self):
```
Comment (kmax12) [inline composeml/label_times.py]: I think a user should be able to provide a `label_type` parameter to override this. Label types could be "categorical" or "continuous". we can also pass this parameter through when using the label maker. for example, bin would know to pass categorical through. 

In terms of infer categorical, I'd also check for any non numeric dtyes like objects, strs, or the pandas categorical dtype

Comment (kmax12) [reply composeml/label_times.py]: additionally, we only want to be calculating this once when the labeltime are initialized, not every time `_is_categorical` is accessed

Comment (jeff-hernandez) [reply composeml/label_times.py]: What do you think about overriding it by using type hints?
```python
def labeling_function(df) -> 'categorical':
    ...
```

Comment (jeff-hernandez) [reply composeml/label_times.py]: Also casting label times to categorical dtype if categorical. Related to #60

Comment (kmax12) [reply composeml/label_times.py]: I don't think that the type hints for annotations  is very intuitive. might just be better to make it a parameter to the label maker

Comment (jeff-hernandez) [reply composeml/label_times.py]: okay, will make parameter for `label_type`

File: composeml/label_times.py [hunk 2]
```diff
@@ -318,3 +315,21 @@ def sample(self, n=None, frac=None, random_state=None):
 
             labels = pd.concat(sample_per_label, axis=0, sort=False)
             return labels
+
+    def infer_type(self):
```
[the remark this answers is not in the document]
Comment (jeff-hernandez) [reply composeml/label_times.py]: updated `infer_type` to return `'discrete'` or `'continuous'`

File: composeml/label_times.py [hunk 3]
```diff
@@ -1,9 +1,10 @@
 import pandas as pd
 
+from composeml.label_plots import LabelPlots
+
 
 class LabelTimes(pd.DataFrame):
```
[the remark this answers is not in the document]
Comment (jeff-hernandez) [reply composeml/label_times.py]: added `label_type` attribute to label times

File: composeml/label_times.py [hunk 4]
```diff
@@ -19,55 +20,50 @@ def __init__(self, data=None, name=None, target_entity=None, settings=None, tran
...
+            value = value.unstack(self.name).fillna(0)
+            value = value.cumsum()
+            return value
+        else:
+            value = self.groupby('cutoff_time')
+            value = value[self.name].count()
+            value = value.cumsum()
+            return value
 
     def describe(self):
```
[the remark this answers is not in the document]
Comment (jeff-hernandez) [reply composeml/label_times.py]: Should label type be under settings?

File: composeml/label_maker.py [hunk 5]
```diff
@@ -267,6 +268,7 @@ def search(self,
             gap (str or int) : Time between examples. Default value is window size.
                 If an integer, search will start on the first event after the minimum data.
             drop_empty (bool) : Whether to drop empty slices. Default value is True.
+            label_type (str) : The label type can be "continuous" or "categorical". Default value is the inferred label type.
```
[the remark this answers is not in the document]
Comment (jeff-hernandez) [reply composeml/label_maker.py]: updated to `discrete`

File: composeml/label_maker.py [hunk 6]
```diff
@@ -327,14 +329,23 @@ def search(self,
 
         labels = LabelTimes(data=labels, name=name, target_entity=self.target_entity)
```
[the remark this answers is not in the document]
Comment (jeff-hernandez) [reply composeml/label_maker.py]: moved check to label times init

File: composeml/label_maker.py [hunk 7]
```diff
@@ -327,25 +327,19 @@ def search(self,
...
-        if label_type is not None:
-            error = 'label type must be "continuous" or "categorical"'
-            assert label_type in ['continuous', 'categorical'], error
-
-            if label_type == 'categorical':
-                labels[labels.name] = labels[labels.name].astype('category')
-
-        else:
-            labels = labels.infer_type()
+        if labels.label_type == 'discrete':
```
Comment (kmax12) [inline composeml/label_maker.py]: can we do this casting before we init the label_times? like above line 330

File: composeml/label_times.py [hunk 8]
```diff
@@ -22,20 +30,44 @@ def __init__(self, data=None, name=None, target_entity=None, settings=None, tran
         self.transforms = transforms or []
         self.plot = LabelPlots(self)
 
+        if label_type is not None:
+            error = 'label type must be "continuous" or "discrete"'
+            assert label_type in ['continuous', 'discrete'], error
+
+        if label_type is None and name in self.columns:
```
Comment (kmax12) [inline composeml/label_times.py]: why would `name` not be in `self.columns`?

File: composeml/label_times.py [hunk 9]
```diff
@@ -22,20 +30,44 @@ def __init__(self, data=None, name=None, target_entity=None, settings=None, tran
...
-        return pd.api.types.is_categorical_dtype(dtype)
+
+        is_discrete = pd.api.types.is_bool_dtype(dtype) \
+            or pd.api.types.is_categorical_dtype(dtype) \
+            or pd.api.types.is_object_dtype(dtype)
+
+        if is_discrete:
+            return True
+
+        labels = self[self.name].iloc[:100]
```
Comment (kmax12) [inline composeml/label_times.py]: for now, let's only look at the dtype to infer type. we can always add this functionality in later

File: composeml/label_times.py [hunk 10]
```diff
@@ -320,16 +355,9 @@ def infer_type(self):
         """Infer label type.
 
         Returns:
-            LabelTimes : Label Times as inferred type.
+            str : Inferred label type. Can be "continuous" or "discrete".
```
Comment (kmax12) [inline composeml/label_times.py]: I think intuitively I'd expect the logic in `is_discrete` to be here in `infer_type` and then I'd expect `is_discrete` to just check if `label_type == "discrete`.

then every we currently check `if self.label_type == 'discrete'` we'd just replace with `is_discrete`

File: composeml/label_maker.py (hunk 7 above)
Comment (jeff-hernandez) [reply composeml/label_maker.py]: Labels are records (list of dictionaries) above line 330. Should I pass records to a pandas data frame to make categorical before initializing label times?

File: composeml/label_times.py (hunk 9 above)
Comment (jeff-hernandez) [reply composeml/label_times.py]: updated function

File: composeml/label_times.py (hunk 10 above)
Comment (jeff-hernandez) [reply composeml/label_times.py]: updated logic

File: composeml/label_maker.py (hunk 7 above)
Comment (kmax12) [reply composeml/label_maker.py]: I see. it's fine to leave it here then

File: composeml/label_times.py (hunk 8 above)
Comment (jeff-hernandez) [reply composeml/label_times.py]: Came across behavior where pandas would initialize label times without passing value for `name`. I refactored logic to infer inside `is_discrete` only if label type is none. [link](https://github.com/FeatureLabs/compose/pull/62/files#diff-5d06a8d5b7b55549065b9c8f90f45edfR47-R48)

**Pull request merged**